File: QE-selenium python/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/selects")
print(driver.title)

select_element = driver.find_element(By.ID, "single-select")
select = Select(select_element)

select.select_by_visible_text("Option 2")
select.select_by_index(2)
select.select_by_value("option4")

options = select.options
for option in options:
    print(option.text)

driver.quit()

File: QE selenium java/Activity16.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import java.util.List;

public class Activity16 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/selects");
        System.out.println(driver.getTitle());

        WebElement selectElement = driver.findElement(By.id("single-select"));
        Select select = new Select(selectElement);

        select.selectByVisibleText("Option 2");
        select.selectByIndex(2);
        select.selectByValue("option4");

        List<WebElement> options = select.getOptions();
        for (WebElement option : options) {
            System.out.println(option.getText());
        }

        driver.quit();
    }
}
